ritasamir/assignment4/CountingIslands.java

```diff
@@ -0,0 +1,69 @@
+package assignment_4;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import assignment_3.Point;
+
+public class CountingIslands {
+	int rowNumber;
+	int columnNumber;
+
+	/**
+	 * @param tiles
+	 *            is a 2-dimensional array of booleans, where false means water
+	 *            and true means land
+	 * @param rowNumber
+	 *            is the number of rows
+	 * @param columnNumber
+	 *            is the number of columns
+	 * @return the number of islands.
+	 */
+	public int getNumberOfIslands(boolean[][] tiles, int rowNumber,
+			int columnNumber) {
+		if (tiles.length != rowNumber || tiles[0].length != columnNumber)
+			throw new Error("the size doesn't match");
+		int counter = 0;
+		this.rowNumber = rowNumber;
+		this.columnNumber = columnNumber;
+		for (int i = 0; i < rowNumber; i++) {
+			for (int j = 0; j < columnNumber; j++) {
+				if (tiles[i][j]) {
+					counter++;

```

Comment:
nit: Maybe add a comment explaining why this works. We know the algorithm already, but if someone who had no idea what the solution was were to read this, they might wonder what's going on. :)